p2panda-store/migrations/20250125065817_create-operations.sql

```diff
@@ -0,0 +1,18 @@
+-- SPDX-License-Identifier: MIT OR Apache-2.0
+
+CREATE TABLE IF NOT EXISTS operations_v1 (
+    hash                    TEXT            NOT NULL    PRIMARY KEY,
+    log_id                  TEXT            NOT NULL,
+    version                 TEXT            NOT NULL,
+    public_key              TEXT            NOT NULL,
+    signature               TEXT            NULL,
```

Comment:
On our `Header` struct signature is optional, but this is only a side-effect of us having a two step approach to constructing the header (eg. add the header fields, then call `sign` to add the signature). I would advocate for our store implementations expecting that a signature will always be present, and if it isn't we return an error on insertion.

Reply:
Good suggestion. I've added a `header.signature.is_none()` check on `insert_operation()` in both store implementations, returning an error if `true`.